test/common/decompressor/zlib_decompressor_impl_test.cc

```diff
@@ -0,0 +1,110 @@
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/hex.h"
+#include "common/compressor/zlib_compressor_impl.h"
+#include "common/decompressor/zlib_decompressor_impl.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Decompressor {
+namespace {
+
+class ZlibDecompressorImplTest : public testing::Test {
+protected:
+  static const int8_t gzip_window_bits{31};
+  static const int8_t memory_level{8};
+};
+
+class ZlibDecompressorImplDeathTest : public ZlibDecompressorImplTest {
+protected:
+  static void decompressorBadInitTestHelper(int8_t window_bits) {
+    ZlibDecompressorImpl decompressor;
+    decompressor.init(window_bits);
+  }
+
+  static void unitializedDecompressorTestHelper() {
+    Buffer::OwnedImpl input_buffer;
+    Buffer::OwnedImpl ouput_buffer;
+    ZlibDecompressorImpl decompressor;
+    TestUtility::feedBufferWithRandomCharacters(input_buffer, 100);
+    decompressor.decompress(input_buffer, ouput_buffer);
+  }
+};
+
+TEST_F(ZlibDecompressorImplDeathTest, DecompressorTestDeath) {
+  EXPECT_DEATH(decompressorBadInitTestHelper(100), std::string{"assert failure: result >= 0"});
+  EXPECT_DEATH(unitializedDecompressorTestHelper(), std::string{"assert failure: result == Z_OK"});
+}
+
+TEST_F(ZlibDecompressorImplTest, CompressDecompressSymetricTesting) {
+  Buffer::OwnedImpl input_buffer;
+  Buffer::OwnedImpl output_buffer;
```

Comment:
I will rename them to compressor_input_buffer & compressor_output_buffer, since compressor output becomes decompressor input later in the test.